index.js

```diff
@@ -4,140 +4,219 @@ window.addEventListener("load", () => {
+        currentString = currentString + numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
       } else {
-        return;
+        currentString = currentString + numButtons[n].textContent;
+        clearCheck();
+        //Else, we will continue to add digits to the currentString, but NOT to the actual display (out of room)
       }
     });
   }
 
   numButtons[10].addEventListener("click", () => {
     if (errLock == true) {
-      return;
+      errLock = false;
+      currentString = "0" + numButtons[10].textContent;
+      numDisplay.textContent = currentString;
+      clearCheck();
     } else if (currentString.includes(".")) {
       return;
+    } else if (currentString == "") {
+      currentString = "0" + numButtons[10].textContent;
+      numDisplay.textContent = currentString;
+      clearCheck();
     } else {
       currentString = currentString + numButtons[10].textContent;
       numDisplay.textContent = currentString;
+      clearCheck();
     }
   });
 
   numButtons[11].addEventListener("click", () => {
-    if (errLock == true || currentString == "" || currentString == ".") {
+    if (errLock == true) {
       return;
-    } else if (currentString.length > 10) {
-      currentString = (Number(currentString) * -1).toExponential(5);
-      numDisplay.textContent = currentString;
+    } else if (currentString == "" && firstNum != "") {
+      currentString = (firstNum * -1).toString();
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+      clearCheck();
     } else {
       currentString = (Number(currentString) * -1).toString();
-      numDisplay.textContent = currentString;
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+      clearCheck();
     }
   });
 
   for (let n = 0; n <= 3; n++) {
     opButtons[n].addEventListener("click", () => {
       if (errLock == true) {
         return;
+      } else if (firstNum != "" && currentOp != "" && currentString != "") {
+        equation();
+        currentOp = opArray[n];
+        clearCheck();
+      } else if (firstNum != "" && currentString == "") {
+        currentOp = opArray[n];
+        clearCheck();
       } else {
         firstNum = Number(currentString);
         currentOp = opArray[n];
         currentString = "";
-        numDisplay.textContent = currentString;
+        clearCheck();
       }
     });
   }
+
   opButtons[4].addEventListener("click", () => {
     if (errLock == true) {
       return;
-    } else if (currentString.includes("-")) {
-      currentString = "Error";
-      numDisplay.textContent = currentString;
+    } else if (
+      Number(currentString) < 0 ||
+      (currentString == "" && firstNum < 0)
```

Comment:
firstNum can be an empty string, so you should check for that before treating it as a number.
```suggestion
      (currentString == "" && firstNum != "" && firstNum < 0)
```